consensus/dbft/dbft.go

```diff
@@ -508,6 +508,24 @@ func New(config *params.DBFTConfig, _ ethdb.Database) (*DBFT, error) {
 
 			return nil
 		}),
+		dbft.WithVerifyBlock(func(b block.Block) bool {
+			ethBlock, ok := b.(*Block)
+			if !ok {
+				return false
+			}
+
```

Comment:
We need to include two extra checks here:
```suggestion
			parent := c.chain.CurrentBlock()
			if parent.Number.Cmp(ethBlock.header.Number) >= 0 {
				log.Warn("proposed block has already outdated",
					"current block number", parent.Number.Uint64(),
					"proposed block number", ethBlock.header.Number)
				return false
			}
			if c.lastTimestamp > ethBlock.header.Time {
				log.Warn("proposed block has small timestamp",
					"ts", ethBlock.header.Time,
					"last", c.lastTimestamp)
				return false
			}
```